### PR TITLE
fix clang-tidy warnings

### PR DESCRIPTION
**Pull request by babaissarkar**
Fixes leftover clang-tidy warnings that causes the CI to fail. (Most probably from #9618)

### CODE REVIEW

File: src/ai/configuration.cpp [hunk 1]
```diff
@@ -111,16 +111,16 @@ void extract_ai_configurations(std::map<std::string, description> &storage, cons
...
 
-void configuration::add_era_ai_from_config(const config &era)
+void configuration::add_era_ai_from_config(const config& era)
 {
 	era_ai_configurations_.clear();
 	extract_ai_configurations(era_ai_configurations_, era);
 }
 
-void configuration::add_mod_ai_from_config(config::const_child_itors mods)
+void configuration::add_mod_ai_from_config(const config::const_child_itors& mods)
```
Comment (gfgtdf) [inline src/ai/configuration.cpp:120]: I think (not fully sure) that config::const_child_itors is a type with the trivial destructor that should usually be passed by value. Can you give more information why it suggests this change?

Comment (Pentarctagon) [reply src/ai/configuration.cpp:120]: Maybe a question for @jj-marr as well?

**Comment by jj-marr** [comment]
I would add that you can try adding `-fix` to `run-clang-tidy` to have it autofix issues.

**Comment by cooljeanius** [comment]
Missed one:
```
1 warning treated as error
src/units/types.cpp:509:126: error: the parameter 'attacks' is copied for each invocation but only used as a const reference; consider making it a const reference [performance-unnecessary-value-param,-warnings-as-errors]
  509 | std::vector<t_string> combine_special_notes(const std::vector<t_string>& direct, const config& abilities, const_attack_itors attacks, const movetype& mt)
      |                                                                                                                              ^
      |                                                                                                           const             &
303 warnings generated.
```

File: src/ai/configuration.cpp (hunk 1 above)
Comment (jj-marr) [reply src/ai/configuration.cpp:120]: `static_assert(std::is_trivially_copyable_v<config::const_child_itors>);` passes on gcc. Does it pass on clang?

Comment (jj-marr) [reply src/ai/configuration.cpp:120]: More specifically, the check itself says:
> The check is only applied to parameters of types that are expensive to copy which means they are not trivially copyable or have a non-trivial copy constructor or destructor.
A trivially copyable type must have a trivial copy constructor and destructor, I believe.
Both clang and gcc seem to identify `boost::iterator_range` correctly as trivially copyable:
https://godbolt.org/z/s8c9oxzGn
Perhaps the better solution is to explicitly whitelist `boost::iterator_range`? There's an `AllowedTypes` option for this check.
https://clang.llvm.org/extra/clang-tidy/checks/performance/unnecessary-value-param.html

**Comment by soliton-** [comment]
I've added a commit to make it a bit easier to see the issues.

File: src/ai/configuration.cpp (hunk 1 above)
Comment (babaissarkar) [reply src/ai/configuration.cpp:120]: > I think (not fully sure) that config::const_child_itors is a type with the trivial destructor that should usually be passed by value. Can you give more information why it suggests this change?

> Perhaps the better solution is to explicitly whitelist boost::iterator_range? There's an AllowedTypes option for this check.

If any other changes are needed, it's best done by @jj-marr in a follow-up PR. I was simply trying to get the broken CI back into working order.

File: src/server/common/dbconn.cpp [hunk 2]
```diff
@@ -586,17 +586,17 @@ bool dbconn::do_any_authors_exist(const std::string& instance_version, const std
 //
 // handle complex query results
 //
-void dbconn::get_complex_results(mariadb::connection_ref connection, rs_base& base, const std::string& sql, const sql_parameters& params)
+void dbconn::get_complex_results(const mariadb::connection_ref& connection, rs_base& base, const std::string& sql, const sql_parameters& params)
 {
-	mariadb::result_set_ref rslt = select(connection, sql, params);
+	mariadb::result_set_ref rslt = select(std::move(connection), sql, params);
```
Comment (Vultraz) [inline src/server/common/dbconn.cpp:591]: This seems wrong. You can't move from a const reference. Bad linter?

Comment (Vultraz) [inline src/server/common/dbconn.cpp:591]: This seems wrong. You can't move from a const reference, and I noticed this same change was applied several times to this file. @jj-marr Is this false linter positive?

Comment (jj-marr) [reply src/server/common/dbconn.cpp:591]: It is wrong. Sometimes the linter gives bad suggestions in this case. I believe you should be able to just mark `const mariadb::connection_ref& connection` and the linter should let it through without the `std::move`.

**Pull request merged**